File: frontend/src/api/health.ts
import { requestJson } from "./client";

export type HealthCheckItem = {
  name: string;
  status: string;
  message: string;
  duration_ms: number | null;
  details: Record<string, unknown>;
};

export type BackendHealth = {
  status: string;
  service: string;
  version: string;
  environment: string;
  checked_at: string;
  storage_root: string;
  checks?: HealthCheckItem[];
  database?: Record<string, unknown>;
  frontends?: Record<string, HealthCheckItem>;
};

export async function getBackendHealth(signal?: AbortSignal): Promise<BackendHealth> {
  return requestJson<BackendHealth>("/health", { signal });
}

File: backend/app/schemas/health.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class HealthCheckItem(BaseModel):
    name: str
    status: str
    message: str
    duration_ms: int | None = None
    details: dict[str, Any] = Field(default_factory=dict)


class HealthResponse(BaseModel):
    status: str
    service: str
    version: str
    environment: str
    checked_at: datetime
    storage_root: str
    checks: list[HealthCheckItem] = Field(default_factory=list)
    database: dict[str, Any] = Field(default_factory=dict)
    frontends: dict[str, Any] = Field(default_factory=dict)
